lib/quicly.c

```diff
@@ -2613,6 +2617,21 @@ static int on_ack_handshake_done(quicly_sentmap_t *map, const quicly_sent_packet
     return 0;
 }
 
+static int on_ack_data_blocked(quicly_sentmap_t *map, const quicly_sent_packet_t *packet, int acked, quicly_sent_t *sent)
+{
+    quicly_conn_t *conn = (quicly_conn_t *)((char *)map - offsetof(quicly_conn_t, egress.loss.sentmap));
+
+    if (conn->egress.max_data.permitted == sent->data.data_blocked.offset) {
```

Comment:
Why is this check necessary? Isn't it enough to see if `conn->egress.data_blocked == QUICLY_SENDER_STATE_SEND`?

Reply:
Because there is no need to send once the flow control is unblocked by the receipt of a MAX_DATA frame. FWIW, in `handle_max_data_frame`, there is a code that turns `conn->egress.data_blocked` to `QUICLY_SENDER_STATE_NONE`.

Reply:
Do you mean BLOCKED frames?

Reply:
There's something I don't follow here. Perhaps we can talk about it in person next week.

Reply:
> Do you mean that we don't need to continue sending BLOCKED frames when MAX_DATA is received?

Correct.